README.md

```diff
@@ -128,6 +128,7 @@ example-gateway                 # root directory
 │   ├── endpoints               # generated mocks and module initializers for endpoints
 │   ├── gen-code                # generated structs and (de)serializers by Thrift compiler
 │   ├── middlewares             # generated module initializers for middlewares
+│   │   └── mandatory           # generated module initializers for mandatory middlewares
```

Comment:
Instead of `mandatory`, I think `default` is a better alternative, since it relates better with `DefaultDepDirs` field in the module class.

Also we should have a section in the doc to explain the concept of default dependencies and how it works.